### PR TITLE
Add test for load of latest snapshot with groupid

### PR DESCRIPTION
**Pull request by jatgarg**
## Description

Add test for load of latest snapshot with groupid and use the latest ISnapshot in container runtime instead of older snapshotTree.

### CODE REVIEW

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -794,9 +794,13 @@ export class ContainerRuntime
...
 
+		const baseSnapshotTree =
+			context.snapshotWithContents !== undefined
+				? context.snapshotWithContents.snapshotTree
+				: context.baseSnapshot;
 		const tryFetchBlob = async <T>(blobName: string): Promise<T | undefined> => {
-			const blobId = context.baseSnapshot?.blobs[blobName];
-			if (context.baseSnapshot && blobId) {
+			const blobId = baseSnapshotTree?.blobs[blobName];
+			if (baseSnapshotTree && blobId) {
```
Comment (vladsud) [inline packages/runtime/container-runtime/src/containerRuntime.ts:803]: I'm curious why this change is needed. I'd think that old code would work, because we would populate both structures.
It might be good long-term, but I'd rather wait for new loader saturation, and then completely remove old code paths (i.e. remove context.baseSnapshot), and with that - find and patch all the places like that.

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 2]
```diff
@@ -1233,10 +1237,13 @@ export class ContainerRuntime
 			loader,
 			pendingLocalState,
 			supportedFeatures,
+			snapshotWithContents,
 		} = context;
 
 		this.innerDeltaManager = deltaManager;
 		this.deltaManager = new DeltaManagerSummarizerProxy(this.innerDeltaManager);
+		const baseSnapshotTree =
```
Comment (vladsud) [inline packages/runtime/container-runtime/src/containerRuntime.ts:1245]: I'd suggest finding better names. baseSnapshot & baseSnapshotTree - not very clear from names how they are different.
Maybe baseSnapshotTree  -> snapshotWithContent?

File: packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNodeUtils.ts [hunk 3]
```diff
@@ -162,7 +162,12 @@ export interface ISubtreeInfo<T extends ISnapshotTree | SummaryObject> {
 export function parseSummaryForSubtrees(baseSummary: ISnapshotTree): ISubtreeInfo<ISnapshotTree> {
 	// New versions of snapshots have child nodes isolated in .channels subtree
 	const channelsSubtree = baseSummary.trees[channelsTreeName];
-	if (channelsSubtree !== undefined) {
+	// If there is ".channels" subtree or there is a groupId with no trees, then that means we use
+	// the format where we store trees inside the ".channels".
+	if (
+		channelsSubtree !== undefined ||
+		(baseSummary.groupId !== undefined && Object.keys(baseSummary.trees).length === 0)
```
Comment (vladsud) [inline packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNodeUtils.ts:169]: I think we should just assert that .channales sub-tree never uses groupId and move on.
In context of that specific question, I think it's fine assumption to have.

The problem with more generic logic - I do not think we can assume that presence groupId tells us anything about the emptiness of the tree. I.e. it could have groupId and be empty. In general, it's not even clear what "empty" means and why someone wants to know that without attempting to load sub-tree.

One more note that is relevant to more general case: We need to be super-clear how we encode things. I see multiple options:
1. Whole tree is always returned irrespective of groupId, only blobs are missing.
2. Sub-tree with groupID could be not returned.
3. Sub-tree with groupID could be is always not returned.

I believe we are not doing # 3. The problem with #2 - we can't answer the question if it's empty or not.
And if it's # 1, then we should not use groupId test at all in code like that.

In either # 1 or 2, we need some indication that tree/blob was not returned:
1. In # 1, presence of groupID and does not tell us if we have all the blobs or not for that sub-tree. We can test it blob by blob, but that's a bit expensive.
2. In # 2, presence of groupID and empty tree does not tell us anything yet if tree exists or not.

Comment (vladsud) [reply packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNodeUtils.ts:169]: I think SPO is planning to do # 1, but we should clearly reflect that in protocol, and I think add some property that says whether sub-tree blobs were included or excluded. Or agree that this is not going to happen, and have helper function that can answer that question by scanning sub-tree blobs

**Review by vladsud — APPROVED** [review]

File: packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNodeUtils.ts (hunk 3 above)
Comment (jatgarg) [reply packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNodeUtils.ts:169]: It would be better if we don't have to add another prop on snapshot tree just to see whether server excluded the blobs or not when we already have one prop groupId for this.

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 1 above)
Comment (jatgarg) [reply packages/runtime/container-runtime/src/containerRuntime.ts]: Removed.

Comment (jatgarg) [reply packages/runtime/container-runtime/src/containerRuntime.ts]: Just wanted to use the new structure.

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 2 above)
Comment (jatgarg) [reply packages/runtime/container-runtime/src/containerRuntime.ts]: Removed as per other comments.
